player/main.c

```diff
@@ -427,6 +427,13 @@ int mpv_main(int argc, char *argv[])
 
     char **options = argv && argv[0] ? argv + 1 : NULL; // skips program name
     int r = mp_initialize(mpctx, options);
+#if !defined(__SANITIZE_ADDRESS__)
+    // allow fast exit on standalone player, unless leak detection is enabled
+    char *ta_report_env = getenv("MPV_LEAK_REPORT");
+    bool ta_report= ta_report_env && ta_report_env[0] == '1' ||
+                    !ta_report_env && HAVE_TA_LEAK_REPORT;
```

Comment:
Honestly, this is pretty ugly.

If you want to do this "properly" I suggest hoisting it out to a function instead of duplicating logic.

```diff
diff --git a/player/main.c b/player/main.c
index 9f20377a6..6d0b088a8 100644
--- a/player/main.c
+++ b/player/main.c
@@ -229,6 +229,14 @@ static int cfg_include(void *ctx, char *filename, int flags)
     return r;
 }
 
+static bool enable_leak_checker_question_mark(void)
+{
+    char *enable_talloc = getenv("MPV_LEAK_REPORT");
+    if (!enable_talloc)
+        enable_talloc = HAVE_TA_LEAK_REPORT ? "1" : "0";
+    return strcmp(enable_talloc, "1") == 0;
+}
+
 // We mostly care about LC_NUMERIC, and how "." vs. "," is treated,
 // Other locale stuff might break too, but probably isn't too bad.
 static bool check_locale(void)
@@ -247,10 +255,7 @@ struct MPContext *mp_create(void)
         return NULL;
     }
 
-    char *enable_talloc = getenv("MPV_LEAK_REPORT");
-    if (!enable_talloc)
-        enable_talloc = HAVE_TA_LEAK_REPORT ? "1" : "0";
-    if (strcmp(enable_talloc, "1") == 0)
+    if (enable_leak_checker_question_mark())
         talloc_enable_leak_report();
 
     mp_time_init();
@@ -429,10 +434,7 @@ int mpv_main(int argc, char *argv[])
     int r = mp_initialize(mpctx, options);
 #if !defined(__SANITIZE_ADDRESS__)
     // allow fast exit on standalone player, unless leak detection is enabled
-    char *ta_report_env = getenv("MPV_LEAK_REPORT");
-    bool ta_report= ta_report_env && ta_report_env[0] == '1' ||
-                    !ta_report_env && HAVE_TA_LEAK_REPORT;
-    mpctx->quit_fast = !ta_report;
+    mpctx->quit_fast = !enable_leak_checker_question_mark();
 #endif
     if (r == 0)
         mp_play_files(mpctx);
```

Reply:
> Honestly, this is pretty ugly.

Really? That's surprising.

Reply:
> Really? That's surprising.

It's duplicating logic around unnecessarily. Also the single byte check isn't the same as `strcmp` which checks for the nul-byte as well.

Reply:
This is why I just preferred removing the leak report altogether.

Reply:
> This is why I just preferred removing the leak report altogether.

I was under impression we are going to do that eventually?

> Also the single byte check isn't the same as strcmp which checks for the nul-byte as well.

It is not, but single byte check is what we need, anyway I created a function and copied exactly old code, I don't mind.

Reply:
>I was under impression we are going to do that eventually?

Well I thought it was a no-brainer PR but apparently it's controversial dunno.